src/jdk.jdwp.agent/share/native/libjdwp/debugInit.c

```diff
@@ -1207,6 +1286,14 @@ parseOptions(char *options)
             if ( !get_boolean(&str, &suspendOnInit) ) {
                 goto syntax_error;
             }
+#if defined(J9VM_OPT_CRIU_SUPPORT)
+            suspendOnRestore = suspendOnRestoreTmp;
```

Comment:
I don't think this does what we want for `suspend=n` (which should imply a default of `suspendOnRestore=n`).

Reply:
This was proposed at https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/591#issuecomment-2296411187
* -agentlib:jdwp=transport=dt_socket,server=y,suspend=[y|n] assumes suspendOnRestore=y

With `-XX:+DebugOnRestore -XX:+EnableCRIUSupport`, the assumption is the user would like to debug post-restore, particularly the post-restore hook application code. Only with `suspendOnRestore=y`, the user has the chance to add breakpoints in those hooks.
If `suspendOnRestore=n` is specified explicitly, it will be honoured and JVM won't be suspended post-restore.

Reply:
I'm not sure that's the behavior we ultimately want, but for now, let's assume it is.

I don't think `suspendOnRestoreTmp` is needed. The assigment on line 1292 does nothing (if `suspendOnRestoreTmp` is true (either because it's the default or the user specified `suspendOnRestore=y`), then `suspendOnRestore` is already true; if `suspendOnRestoreTmp` is false, it's because the user said `suspendOnRestore=n` in which case `suspendOnRestore` is already false. Line 1292 can simply be deleted, and line 1294 should use `suspendOnRestore`.

Reply:
Yeah, the temp isn't needed, removed it.